kafka/consumer-group.go

```diff
@@ -32,25 +34,38 @@ func (cg ConsumerGroup) Errors() chan error {
 	return cg.errors
 }
 
+// Release signals that upstream has completed an incoming message
+// i.e. move on to read the next message
+func (cg ConsumerGroup) Release() {
+	cg.upstreamDone <- true
+}
+
+// Release signals the successful completion of a read incoming message
+func (cg ConsumerGroup) CommitAndRelease(msg Message) {
+	log.Trace("pre-commit", nil)
+	msg.Commit()
+	log.Trace("postCommit pre-release", nil)
+	cg.Release()
+	log.Trace("           postRelease", nil)
+}
+
 // StopListeningToConsumer stops any more messages being consumed off kafka topic
 func (cg *ConsumerGroup) StopListeningToConsumer(ctx context.Context) (err error) {
 
 	if ctx == nil {
 		ctx = context.Background()
 	}
 
-	var q struct{}
-	cg.closer <- q
+	close(cg.closer)
 
 	select {
 	case <-cg.closed:
-		log.Info("Stopped listening to kafka consumer group", log.Data{"topic": cg.topic})
-		return
+		log.Info("StopListeningToConsumer saw closed kafka consumer", log.Data{"topic": cg.topic, "group": cg.group})
```

Comment:
The message doesn't make sense to me, Im not sure what the problem was with the previous log message?

Reply:
I changed the comment again